nodejs/src/services/AuthService.ts

```diff
@@ -33,7 +33,7 @@ export class AuthService {
     type: TokenType,
     { issuedAt, notValidBefore, validForSeconds }: BaseTokenConfig = {},
     extras: Record<string, any> = {}
-  ): Promise<ManagementToken> {
+  ): Promise<Token> {
```

Comment:
I think we can overload this perhaps for better types, wdyt?

```ts
private async generateToken(type: TokenType.App, config: BaseTokenConfig, extras: Record<string, any>): AuthToken
private async generateToken(type: TokenType. Management, config: BaseTokenConfig, extras: Record<string, any>): ManagementToken
private async generateToken(
    type: TokenType,
    { issuedAt, notValidBefore, validForSeconds }: BaseTokenConfig = {},
    extras: Record<string, any> = {}
  ): Promise<ManagementToken | AuthToken> {
  ): Promise<Token> {
//.... implementation
}
```

Reply:
https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads